src/structures/Webhook.js

```diff
@@ -235,18 +237,40 @@ class Webhook {
     return this;
   }
 
+  /**
+   * Options that can be passed into fetchMessage.
+   * @typedef {options} WebhookFetchMessageOptions
+   * @property {boolean} [cache=true] Whether to cache the message.
+   * @property {Snowflake} [threadId] The id of the thread this message belongs to.
+   * <info>For interaction webhooks, this property is ignored</info>
+   */
+
   /**
    * Gets a message that was sent by this webhook.
    * @param {Snowflake|'@original'} message The id of the message to fetch
-   * @param {boolean} [cache=true] Whether to cache the message
+   * @param {WebhookFetchMessageOptions|boolean} [cacheOrOptions={}] The options to provide to fetch the message.
+   * A **deprecated** boolean may be passed instead to specify whether to cache the message.
    * @returns {Promise<Message|APIMessage>} Returns the raw message data if the webhook was instantiated as a
    * {@link WebhookClient} or if the channel is uncached, otherwise a {@link Message} will be returned
    */
-  async fetchMessage(message, cache = true) {
+  async fetchMessage(message, cacheOrOptions = { cache: true }) {
+    if (typeof cacheOrOptions === 'object') {
+      cacheOrOptions.cache ??= true;
+    } else {
+      cacheOrOptions = { cache: cacheOrOptions };
+    }
```

Comment:
Not a fan of mutating objects, we can do this instead:
```suggestion
    if (typeof cacheOrOptions === 'boolean') {
      cacheOrOptions = { cache: cacheOrOptions };
    }
```

We don't need to default `options.cache` to `true`, because it's defaulted to that same value in the manager:

https://github.com/discordjs/discord.js/blob/3b14883e349c23a15aec225bb133ce49ed440817/src/managers/CachedManager.js#L46